stdlib/lazy.mli

```diff
@@ -42,9 +42,11 @@ type 'a t = 'a CamlinternalLazy.t
    for the [lazy] keyword.  You should not use it directly.  Always use
    [Lazy.t] instead.
 
-   Note: [Lazy.force] is not thread-safe.  If you use this module in
-   a multi-threaded program, you will need to add some locks.  See also
-   [Lazy.try_force].
+   Note: [Lazy.force] (and therefore the [lazy] pattern-matching) are
+   thread-safe, but will raise the [Undefined] exception if forced
+   concurrently. If you need to share a lazy between threads, then you
+   need to use [Lazy.try_force] and implement your own synchronisation.
```

Comment:
I was not sure about the terminology: "implement your own synchronisation" -> "implement your own locking mechanism"? What do you think?

Reply:
`synchronization` the right word. Perhaps we could add the example of:

```ocaml
let rec conc_force l =
  match Lazy.try_force with
  | None -> Domain.Sync.cpu_relax(); conc_force l
  | Some v -> v
```

in the documentation as an example of synchronization where the concurrent access spin-waits until completion.

Reply:
I am wondering whether this could be offered in a library. It seems fine to offer it if the user has to make a conscious decision to use it.

Reply:
Ok. That's fine. Let's mark this one as resolved.